src/ts/background/handlers/subscriptions.ts

```diff
@@ -0,0 +1,29 @@
+import { Runtime } from 'webextension-polyfill-ts'
+
+type Subscriptions = {
+  [index: string]: Runtime.Port
+}
+
+const subscriptions: Subscriptions = {}
+
+// return a subscription callback, that will send the data to the caller via the port
+export function createSubscription (id: string, port: Runtime.Port): (data: any) => void {
+  subscriptions[id] = port
+
+  return (subscription: any) => {
+    if (subscriptions[id]) {
```

Comment:
is this check required as we just set its value above?

Reply:
yes, I think so. It's checking if the subscriber (extension popup or web page) is still connected. It will post message only if the subscriber is still connected.